lib/plugins/aws/package/lib/generateCoreTemplate.js

```diff
@@ -21,6 +21,16 @@ module.exports = {
       )
     );
 
+    const { region } = this.options;
+    if (region.startsWith('cn-')) {
+      // Ensure correct ARN for china region
+      const arnTokens = this.serverless.service.provider.compiledCloudFormationTemplate.Resources
+        .ServerlessDeploymentBucketPolicy.Properties.PolicyDocument.Statement[0].Resource[0][
+        'Fn::Join'
```

Comment:
IMHO we should use the partition pseudo parameter here (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-partition). So basically we need to update the [core template](https://github.com/serverless/serverless/blob/b39c10c8d213168c376cf5fde9d28af5d0c1974d/lib/plugins/aws/package/lib/core-cloudformation-template.json#L33).

If you're searching for `AWS::Partition` you'll find some examples in the code.

Reply:
@pmuens indeed, very good and important point. Honestly I didn't like the initial solution, didn't seem that futureproof, was not aware of `AWS::Partition`.

Updated